Adds section for AEM Assets integration in merchant docs (#482)
commerce-docs/microsite-commerce-storefront: src/content/docs/setup/configuration/aem-assets-configuration.mdx

---
title: AEM Assets integration
description: Learn how to connect your Edge Delivery Services storefront to your AEM Assets integration.
---

import OptionsTable from '@components/OptionsTable.astro';
import Tasks from '@components/Tasks.astro';
import Task from '@components/Task.astro';
import Vocabulary from '@components/Vocabulary.astro';
import { Tabs, TabItem, Aside } from '@astrojs/starlight/components';

The AEM Assets integration allows your storefront to display product images that are managed in AEM Assets instead of traditional Commerce-hosted images. This integration provides enhanced image management capabilities, including advanced optimization, cropping, and delivery through Adobe's Content Delivery Network (CDN).

## Boilerplate update required

> Only one update is required: set `"commerce-assets-enabled": true` in your `config.json` file, renamed from `/demo-config.json`. See [Storefront configuration](/setup/configuration/commerce-configuration/) for the full configuration model.

Reviewer: We should not be pushing people to `demo-config.json` or `config.json` in repo. Just link to the configuration page.

```suggestion
Only one update is required: set `"commerce-assets-enabled": true` in your [Storefront configuration](/setup/configuration/commerce-configuration/).
```

```json title="config.json" ins={'+':4}
{
  "public": {
    "default": {
      "commerce-assets-enabled": true
    }
  }
}
```

> The Commerce drop-ins automatically detect the `commerce-assets-enabled` configuration and adjust image handling accordingly. See the [demo-config.json](https://github.com/hlxsites/aem-boilerplate-commerce/blob/main/demo-config.json) file in the boilerplate repository. 

Reviewer: Do not link to this file. Reference the live configuration on aemshop.net at most. https://www.aemshop.net/config.json
```suggestion
The Commerce drop-ins automatically detect the `commerce-assets-enabled` configuration and adjust image handling accordingly. See the [demo configratuion](https://www.aemshop.net/config.json). 
```

<OptionsTable
  centerColumns={[1]}
  options={[
    ['Image source', 'commerce-assets-enabled setting', 'Expected behavior'],
    ['Commerce-hosted images', 'true', 'Images display correctly. The AEM Assets integration code passes through Commerce-hosted images without modification.'],
    ['AEM Assets images', 'true', 'Images display correctly with proper AEM Assets CDN optimization parameters applied. This is the intended configuration for AEM Assets integration.'],
    ['Commerce-hosted images', 'false', 'Images display correctly using standard Commerce image handling without AEM Assets optimization.'],
    ['AEM Assets images', 'false', 'Images may not display correctly. AEM Assets images require specific optimization parameters that may conflict with standard Commerce image handling, potentially resulting in 400 errors or broken images.'],
  ]}
/>

<Aside type="tip" title="Compatibility notes">

> - Commerce-hosted images continue to work when AEM Assets is enabled.

Reviewer: What does this mean? How would it work?

Reviewer 2: @sirugh It's my understanding that the assets.js passes through commerce-stored images first before it accesses the aem assets images.

> - If you're not using the boilerplate, add AEM support in the drop-in image slots with `tryRenderAemAssetsImage`.

Reviewer: Is this function actually usable outside of the boilerplate? I think we expose it from tools (tools/lib/aem/assets.js) along with other functions. Might be worth a separate page or section specifically about using it (if we want to expose that).

From assets.d.ts: 
```
/** Returns whether AEM Assets is enabled in the Storefront. */
export declare function isAemAssetsEnabled(): boolean;
/** The default optimization parameters used globally, unless overriden (per use). */
export declare function getDefaultAemAssetsOptimizationParams(): AemAssetsParams;
/** Returns true if the given URL is an AEM Assets URL. */
export declare function isAemAssetsUrl(url: string | URL): boolean;
/** Generates an optimized URL for AEM Assets. */
export declare function generateAemAssetsOptimizedUrl(assetUrl: string, alias: string, params?: Partial<AemAssetsParams>): string;
/**
 * Tries to generate an optimized URL for AEM Assets. Returns the given
 * url if AEM Assets is not enabled or is not an AEM Assets URL.
 */
export declare function tryGenerateAemAssetsOptimizedUrl(assetUrl: string, alias: string, params?: Partial<AemAssetsParams>): string;
/** Creates a slot that renders an AEM Assets image. */
export declare function makeAemAssetsImageSlot(config: AemAssetsImageSlotConfig): (ctx: RenderContext) => void;
export declare function tryRenderAemAssetsImage(ctx: RenderContext, config: AemAssetsImageSlotConfig): void;
export {};
//# sourceMappingURL=assets.d.ts.map
```

Reviewer: Edit: I see that you have details below about how it is used. Still might be worth noting the other functions?

  - `isAemAssetsUrl(url)`
  - `generateAemAssetsOptimizedUrl(assetUrl, alias, params)`
  - `tryGenerateAemAssetsOptimizedUrl(assetUrl, alias, params)`
  - `makeAemAssetsImageSlot(config)`
  - `tryRenderAemAssetsImage(ctx, config)`

</Aside>

## Troubleshooting

### Images not displaying

If product images are not displaying correctly:

1. **Check your configuration:** Ensure `commerce-assets-enabled` is set to `true` if you're using AEM Assets.
2. **Verify image URLs:** AEM Assets images typically include specific URL patterns that indicate they're served from AEM.
3. **Check browser console:** Look for 400 errors that might indicate incompatible optimization parameters.
4. **Test with mixed sources:** Try with both AEM Assets and Commerce-hosted images to isolate the issue.
5. **Confirm slot integration:** If not using the boilerplate, ensure each drop-in image slot calls `tryRenderAemAssetsImage` with a valid `alias` (SKU).
6. **Validate size params:** Ensure `params.width`/`params.height` match the slot's intended render size (for example, from `defaultImageProps`).

### Configuration not taking effect

1. **Clear cache:** Ensure your configuration changes are not cached.

> 2. **Check config location:** Verify your `config.json` is in the root directory of your project (same level as `package.json`) and properly formatted. Alternatively, you can use the [config service](https://www.aem.live/docs/config-service-setup) where your config lives at `https://admin.hlx.page/config/{{ORG}}/sites/{{SITE}}/public.json`.

Reviewer: ```suggestion
2. **Check config location:** Verify your [storefront configuration](https://experienceleague.adobe.com/developer/commerce/storefront/setup/configuration/commerce-configuration/) is properly formatted.
```

import Task from '@components/Task.astro';

In this section, you'll learn how Commerce blocks in your storefront connect to an AEM Assets integration using values from either your site's [public config](https://www.aem.live/docs/admin.html#schema/PublicConfig) or a [config.json](https://github.com/hlxsites/aem-boilerplate-commerce/blob/main/demo-config.json) file in your code repo.

You'll need to add the property `commerce-assets-enabled` to your `config.json` file to enable compatibility with the AEM Assets integration.

```json title="config.json" showLineNumbers ins={'+':4}
{
  "public": {
    "default": {
      "commerce-assets-enabled": true,
      "commerce-core-endpoint": "https://www.aemshop.net/graphql",
      "commerce-endpoint": "https://www.aemshop.net/cs-graphql",
    }
  }
}
```

<Callouts>

1. **commerce-assets-enabled:** (read/write) Enables the use of AEM Assets integration in your storefront. See [AEM Assets Integration](https://experienceleague.adobe.com/en/docs/commerce/aem-assets-integration/overview) documentation for details.

</Callouts>

If you [set up your storefront](/get-started/) using the boilerplate code, AEM Assets is supported out-of-the-box. 

> If you set up your storefront without the boilerplate code, you can still enable compatibility with the AEM Assets integration by configuring each image slot individually.

Reviewer 3: Are these images slots documented individually? As all the others are for the different dropins? Many were newly created when implementing this feature.

Reviewer 2: Updated with sections on 'How it works in the boilerplate' and 'Using drop-ins outside the boilerplate (optional)'

## Expected behaviors

> Find below what is expected to happen in each different configuration combination:

Reviewer 3: Is the expectation for merchants to know what "different configuration combination" means? Like the table speaks of "Source" and "Type" I can see, but is it explained what is that? 

"Source" should be the Visualization Owner Setting of the AEM Assets Integration in Commerce
"Type" should be whether you have the `commerce-assets-enabled` to `true` or `false`

Reviewer 2: Done. I clarified the meaning of the table columns headers.

<OptionsTable
  options={[
    ['Source', 'Type', 'Description'],

>     ['Commerce', 'true', 'Only hosted images will be returned from the APIs, but config is indicating that images will come from AEM Assets, the code should not fail as the integration code should passthrough. This only applies for drop-in slots, not widgets compatibility workarounds.'],

Reviewer 3: Remove the "_this only applies for drop-in slots, not widgets compatibility workarounds_". That line was initially referring to workarounds we applied to widget code in the boilerplate before PLP became a dropin. Don't think it's relevant anymore as now everything is done via dropins.

Reviewer 2: Done
